SDWebImage/Core/SDImageIOAnimatedCoder.m

```diff
@@ -265,20 +339,8 @@ - (UIImage *)incrementalDecodedImageWithOptions:(SDImageCoderOptions *)options {
     
     if (_width + _height > 0) {
         // Create the image
-        CGImageRef partialImageRef = CGImageSourceCreateImageAtIndex(_imageSource, 0, NULL);
-        
-        if (partialImageRef) {
-            CGFloat scale = _scale;
-            NSNumber *scaleFactor = options[SDImageCoderDecodeScaleFactor];
-            if (scaleFactor != nil) {
-                scale = MAX([scaleFactor doubleValue], 1);
-            }
-#if SD_UIKIT || SD_WATCH
-            image = [[UIImage alloc] initWithCGImage:partialImageRef scale:scale orientation:UIImageOrientationUp];
-#else
-            image = [[UIImage alloc] initWithCGImage:partialImageRef scale:scale orientation:kCGImagePropertyOrientationUp];
-#endif
-            CGImageRelease(partialImageRef);
+        image = [SDImageIOAnimatedCoder createFrameAtIndex:0 source:_imageSource scale:_scale preserveAspectRatio:_preserveAspectRatio thumbnailSize:_thumbnailSize];
+        if (image) {
```

Comment:
Maybe you had forgot `SDImageCoderDecodeScaleFactor `?

Reply:
@kinarobin This is the progressive decoding. The `_scale` is initialized in `initIncrementalWithOptions:` method. See above 281line.

Reply:
Oh....I forgot this edge case. If you pass `SDImageCoderDecodeScaleFactor ` in both `initIncrementalWithOptions:`, and also this `incrementalDecodedImageWithOptions:`. We prefer to use this method arg firstlly, then backup with `initIncrementalWithOptions:`... Fix it